controllers/resource/resource_generator.go

```diff
@@ -46,57 +46,81 @@ type limitPatcher interface {
 	PatchDeployment(deployment *appsv1.Deployment, limits limit.DoguLimits) error
 }
 
-// GetDoguDeployment creates a new instance of a deployment with a given dogu.json and dogu custom resource
-func (r *ResourceGenerator) GetDoguDeployment(doguResource *k8sv1.Dogu, dogu *core.Dogu, customDeployment *appsv1.Deployment) (*appsv1.Deployment, error) {
+func (r *ResourceGenerator) getDoguLabels(doguResource *k8sv1.Dogu) map[string]string {
+	return map[string]string{"dogu": doguResource.Name}
+}
+
+// GetPodTemplate returns a pod template for the given dogu. If the support mode flag is set to true, the pod template
+// receives a custom command which turns the pod to a sleep infinity mode.
+func (r *ResourceGenerator) GetPodTemplate(doguResource *k8sv1.Dogu, dogu *core.Dogu, supportMode bool) *corev1.PodTemplateSpec {
```

Comment:
The method `GetPodTemplate()` is new and will be used from the `GetDoguDeployment()` method (extension).
In the process the signature of `GetDoguDeployment()` has not changed (no change).

Your expected change to edit the supportmode stuff from the template in the `doguSupportManager` sounds handy but is not easily possible. You can't change probes or container from a PodTemplate. They are immutable. You always have to create a new PodTemplate. For this template you need other logic like the creation of the volumes or volumemounts. So either you have make these methods public to create the template in the `doguSupportManager` or you put the creation in the `resourceGenerator`.

I used to select the second implementation because in my opinion the `resourceGenerator` is a good place to handle this. We have resources in different states (supportMode) why he should not able to generate those resources?